tests/integration/test_agentd/test_agentd_reconnection/test_agentd_connection_retries_pre_enrollment.py

```diff
@@ -0,0 +1,145 @@
+configs_path = Path(CONFIGS_PATH, 'wazuh_conf.yaml')
+cases_path = Path(TEST_CASES_PATH, 'cases_reconnection_protocol.yaml')
+
+# Test configurations.
+config_parameters, test_metadata, test_cases_ids = get_test_cases_data(cases_path)
+test_configuration = load_configuration_template(configs_path, config_parameters, test_metadata)
+
+if sys.platform == WINDOWS:
+    local_internal_options = {AGENTD_WINDOWS_DEBUG: '2'}
+else:
+    local_internal_options = {AGENTD_DEBUG: '2'}
+local_internal_options.update({AGENTD_TIMEOUT: '5'})
+
+# Tests
+@pytest.mark.parametrize('test_configuration, test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_agentd_connection_retries_pre_enrollment(test_metadata, set_wazuh_configuration, configure_local_internal_options, truncate_monitored_files):
+    '''
+    description: Check how the agent behaves when the 'wazuh-remoted' daemon is not available
+                 and performs multiple connection attempts to it. For this, the agent starts
+                 with keys but the 'wazuh-remoted' daemon is not available for several seconds,
+                 then the agent performs multiple connection retries before requesting a new enrollment.
+
+                 This test covers and check the scenario of Agent starting with keys but Remoted is not 
+                 reachable during some seconds and multiple connection retries are required prior to 
+                 requesting a new enrollment
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - test_metadata:
+            type: data
+            brief: Configuration cases.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Set internal configuration for testing.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Reset the 'ossec.log' file and start a new monitor.
+
+    assertions:
+        - Verify that the agent enrollment is successful.
+
+    input_description: An external YAML file (wazuh_conf.yaml) includes configuration settings for the agent.
+                       Two test cases are found in the test module and include parameters
+                       for the environment setup using the TCP and UDP protocols.
+
+    expected_output:
+        - r'Sending keep alive'
+
+    tags:
+        - simulator
+        - ssl
+        - keys
+    '''
+    # Stop target Agent
+    control_service('stop')
+
+    # Add dummy key in order to communicate with RemotedSimulator
+    add_custom_key()
+    
+    # Start service
+    control_service('start')
+
+    # Start RemotedSimulator
+    remoted_server = RemotedSimulator(protocol = test_metadata['PROTOCOL'])
+    remoted_server.start()
+    
+    # Start hearing logs
+    wait_keepalive()
+
+    # Reset simulator
+    kill_server(remoted_server)
+    
```

Comment:
Fix end of file. Repeat where needed.

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/18353/commits/ecf23d03eafffc97b67a2d364324e916eb50d143